src/dataflow-types/src/client.rs

```diff
@@ -961,28 +1018,204 @@ pub mod process_local {
+    /// stream coming back. Both of these streams are backed by one tokio mpsc queue each.
+    /// The recv and send functions interact with those two mpsc channels to send and receive data.
+    #[derive(Debug)]
+    pub struct GrpcComputedClient {
+        addr: String,
+        state: GrpcComputedTcpConn,
+        backoff: Duration,
+    }
+
+    /// The connection state of the GrpcComputedClient.
+    #[derive(Debug)]
+    enum GrpcComputedTcpConn {
+        // Initial, disconnected state
+        Disconnected,
+
+        // We have a TCP client connection, but we have to wait for RPC answer for setting up the streams
+        AwaitResponse(ProtoComputeClient<tonic::transport::Channel>),
+
+        // Ready to go!
+        Connected(
+            (
+                Sender<super::ProtoComputeCommand>,
+                Streaming<super::ProtoComputeResponse>,
+            ),
+        ),
+
+        // Unable to connect, wait on next connect
+        Backoff(Instant),
+    }
+
+    impl FromAddr for GrpcComputedClient {
+        fn from_addr(addr: String) -> GrpcComputedClient {
+            GrpcComputedClient {
+                addr: format!("http://{}", addr),
+                state: GrpcComputedTcpConn::Disconnected,
+                backoff: Duration::from_millis(10),
+            }
+        }
+    }
+
+    impl GrpcComputedClient {
+        // This is and must be cancellation safe
+        pub async fn connect(&mut self) -> () {
+            loop {
+                match &mut self.state {
+                    GrpcComputedTcpConn::Disconnected => {
+                        info!("Attempt to connect to {}", &self.addr);
+                        match ProtoComputeClient::connect(self.addr.clone()).await {
+                            Ok(client) => {
+                                self.backoff = Duration::from_millis(10);
+                                self.state = GrpcComputedTcpConn::AwaitResponse(client);
+                            }
+                            Err(e) => {
+                                self.backoff = cmp::min(self.backoff * 2, Duration::from_secs(1));
+                                info!(
+                                    "Connection refused: {}. Backoff {}ms",
+                                    e,
+                                    self.backoff.as_millis()
+                                );
+                                self.state =
+                                    GrpcComputedTcpConn::Backoff(Instant::now() + self.backoff);
+                            }
+                        }
+                    }
+                    GrpcComputedTcpConn::AwaitResponse(client) => {
+                        // The channel size is a arbitrary, but it should be a
+                        // small bounded channel such that backpressure is applied early.
+                        let (tx, rx) = mpsc::channel(4);
+                        match client.handle(ReceiverStream::new(rx)).await {
```

Comment:
does `handle` require a `'static` stream?

Reply:
I dont know how to check this and what it would mean :confused:  The signature of the generated code is 
```
       pub async fn handle(                                                                                                                                                                                                                                                                                                                             
              &mut self,                                                                                                                                                                                                                                                                                                                                   
              request: impl tonic::IntoStreamingRequest<                                                                                                                                                                                                                                                                                                   
                  Message = super::ProtoComputeCommand,                                                                                                                                                                                                                                                                                                    
              >,                                                                                                                                                                                                                                                                                                                                           
          )
```